fs/memoryfs.py

```diff
@@ -137,6 +137,14 @@ def readable(self):
         # type: () -> bool
         return self._mode.reading
 
+    def readinto(self, b):
+        # type (bytes) -> Optional[int]
```

Comment:
I think the typing should be `(bytearray) -> int`.

Although, the docs do say this [method](https://docs.python.org/3/library/io.html#io.RawIOBase.readinto) accepts a "bytes-like" object, but I don't know how to type that.

Reply:
I chose `bytes` for this because of this note in the `typing` docs: https://docs.python.org/3/library/typing.html#typing.ByteString

> class typing.ByteString(Sequence[int])
>    A generic version of collections.abc.ByteString.
>
>    This type represents the types bytes, bytearray, and memoryview.
>
>    As a shorthand for this type, bytes can be used to annotate arguments of any of the types mentioned above.

I think a `memoryview` is also acceptable input to `readinto`.

An alternative to `bytes` here should be `typing.ByteString` (I think :)).

Reply:
It can't be bytes, because readinto needs a mutable object.

The gold standard is probably typeshed (https://github.com/python/typeshed/blob/master/stdlib/3/io.pyi#L55) which uses bytearray.